lua/gitlab/utils/init.lua

```diff
@@ -515,12 +524,37 @@ end
 ---@field new_line integer
 ---@field new_range integer
 
+---@class HunksAndDiff
+---@field hunks Hunk[] list of hunks
+---@field all_diff_output table The data from the git diff command
+
+---Turn hunk line into Lua table
+---@param line table
+---@return Hunk|nil
+M.parse_possible_hunk_headers = function(line)
+  if line:sub(1, 2) == "@@" then
+    -- match:
+    --  @@ -23 +23 @@ ...
+    --  @@ -23,0 +23 @@ ...
+    --  @@ -41,0 +42,4 @@ ...
+    local old_start, old_range, new_start, new_range = line:match("@@+ %-(%d+),?(%d*) %+(%d+),?(%d*) @@+")
+
+    return {
+      old_line = tonumber(old_start),
+      old_range = tonumber(old_range) or 0,
```

Comment:
Hi @harrisoncramer, first of all, thank you for all your work on this - the more I see of the code of `gitlab.nvim` the more I admire what you've achieved and given to the Neovim community.

I think this line should be changed like this:
```suggestion
      old_range = tonumber(old_range) or 1,
```
Here's why: I've been trying to break the new way one-line comments are created and it seems I've found at least one way. In branch `testing1` I've got `file1.txt` with the following content:
```
abc
```
In branch `testing2`, the content of `file1.txt` is changed to this:
```
def
ghi
```
I'm trying to merge `testing2` into `testing1`. The diff hunk looks like this:
```
diff --git a/file.txt b/file.txt
index 8baef1b..aff48dc 100644
--- a/file.txt
+++ b/file.txt
@@ -1 +1,2 @@
-abc
+def
+ghi
```
Now, if I try to put a comment on line 1 in the "new" window ("def"), I'm getting the following error:
```
gitlab.nvim: Could not create discussion: POST https://gitlab.cloud.phonexia.com//api/v4/projects/459/merge_requests/4/discussions: 400 {message: 400 Bad request - Note {:line_code=>["can't be blank", "must be a valid line code
"]}}
```
I've tried to debug this and the problem seems to be that if I delete one line and replace it with two (or more) lines, `old_range` is calculated as `0` and in the loop starting at line 418 in `lua/gitlab/reviewer/diffview.lua`, the `hunk_line_index` is off by `1`, and the "def" is not recognized as `"added"`, because what we are actually matching against `"^%+"` is `"-abc"` and the resulting `modification_type` is determined as `"bad_file_unmodified"`. The following addition `"ghi"` is then recognized as `"added"`, because `"^%+"` is matched against `"+def"`.

I'm not sure if `old_range = tonumber(old_range) or 1` doesn't break anything else, but it certainly fixes the error I just described, and I believe you should be able to reproduce easily.

Reply:
Hey, thanks for catching this.

Your solution worked well but I needed to account for one other edge case. I pushed up that fix. We only want to set that number to 1 when the old range is zero AND the old line is zero as well. I think this fixes everything...

Reply:
Thanks @jakubbortlik and @harrisoncramer for fixing this.